src/components/structures/EmbeddedPage.js

```diff
@@ -82,19 +84,31 @@ export default class EmbeddedPage extends React.PureComponent {
                 this.setState({ page: body });
             },
         );
+
+        this._dispatcherRef = dis.register(this.onAction);
     }
 
     componentWillUnmount() {
         this._unmounted = true;
+        dis.unregister(this._dispatcherRef);
     }
 
+    onAction = (payload) => {
+        // HACK: Workaround for the context's MatrixClient not being set up at render time.
+        if (payload.action === 'client_started') {
+            this.forceUpdate();
+        }
+    };
+
     render() {
-        const client = this.context.matrixClient;
+        // HACK: Workaround for the context's MatrixClient not updating.
+        const client = this.context.matrixClient || MatrixClientPeg.get();
         const isGuest = client ? client.isGuest() : true;
         const className = this.props.className;
         const classes = classnames({
             [className]: true,
             [`${className}_guest`]: isGuest,
+            [`${className}_loggedIn`]: !!client,
```

Comment:
This is fine, but it does make me wonder if we should have a very high level class about logged in set on some top level component in the app? Then the welcome page and anything else can use CSS based on that one class, if it wants to, rather than sprinkling logged in classes to specific components like this. Anyway, up to you.

Reply:
I'm fairly uncomfortable with the amount of indirection already here (it's not obviously clear that these classes are available). Putting them higher feels like a recipe for subtle and nasty bugs.